exchanges/btse/btse_test.go

```diff
@@ -542,80 +550,25 @@ func TestMatchType(t *testing.T) {
-	t.Parallel()
-	seedOrderSizeLimitMap()
-	_, ok := OrderSizeLimits(spotPair.String())
-	assert.True(t, ok, "OrderSizeLimits should find BTC-USD")
-
-	_, ok = OrderSizeLimits("XRP-GARBAGE")
-	assert.False(t, ok, "OrderSizeLimits should not find XRP-GARBAGE until the next bull market")
-}
-
-func seedOrderSizeLimitMap() {
-	testOrderSizeLimits := []struct {
-		name string
-		o    OrderSizeLimit
-	}{
-		{
-			name: "XRP-USD",
-			o: OrderSizeLimit{
-				MinSizeIncrement: 1,
-				MinOrderSize:     1,
-				MaxOrderSize:     1000000,
-			},
-		},
-		{
-			name: "LTC-USD",
-			o: OrderSizeLimit{
-				MinSizeIncrement: 0.01,
-				MinOrderSize:     0.01,
-				MaxOrderSize:     5000,
-			},
-		},
-		{
-			name: "BTC-USD",
-			o: OrderSizeLimit{
-				MinSizeIncrement: 0.0001,
-				MinOrderSize:     1,
-				MaxOrderSize:     1000000,
-			},
-		},
-	}
-
-	orderSizeLimitMap.Range(func(key interface{}, _ interface{}) bool {
-		orderSizeLimitMap.Delete(key)
-		return true
-	})
-
-	for x := range testOrderSizeLimits {
-		orderSizeLimitMap.Store(testOrderSizeLimits[x].name, testOrderSizeLimits[x].o)
-	}
-}
-
-func TestWithinLimits(t *testing.T) {
+// TestUpdateOrderExecutionLimits exercises UpdateOrderExecutionLimits
+func TestUpdateOrderExecutionLimits(t *testing.T) {
 	t.Parallel()
 	testexch.UpdatePairsOnce(t, b)
-	seedOrderSizeLimitMap()
-	p, _ := currency.NewPairDelimiter("XRP-USD", "-")
-	assert.NoError(t, b.withinLimits(p, 1.0), "withinLimits should not error")
-	assert.NoError(t, b.withinLimits(p, 5.0000001), "withinLimits should not error")
-	assert.NoError(t, b.withinLimits(p, 100), "withinLimits should not error")
-	assert.NoError(t, b.withinLimits(p, 10.1), "withinLimits should not error")
-
-	p.Base = currency.LTC
-	assert.NoError(t, b.withinLimits(p, 10), "withinLimits should not error")
-	assert.ErrorIs(t, b.withinLimits(p, 0.009), order.ErrAmountBelowMin, "withinLimits should error correctly")
-
-	p.Base = currency.BTC
-	assert.NoError(t, b.withinLimits(p, 10), "withinLimits should not error")
-	assert.ErrorIs(t, b.withinLimits(p, 0.001), order.ErrAmountBelowMin, "withinLimits should error correctly")
+	for _, a := range b.GetAssetTypes(false) {
+		err := b.UpdateOrderExecutionLimits(context.Background(), a)
+		require.NoErrorf(t, err, "UpdateOrderExecutionLimits must not error for %s", a)
+
+		pairs, err := b.GetAvailablePairs(a)
+		require.NoErrorf(t, err, "GetAvailablePairs must not error for %s", a)
+		require.NotEmpty(t, pairs, "GetAvailablePairs must return some pairs")
+		p := pairs[0]
+
+		limits, err := b.GetOrderExecutionLimits(a, p)
+		require.NoErrorf(t, err, "GetOrderExecutionLimits must not error for %s %s", a, p)
+		assert.Positivef(t, limits.MinimumBaseAmount, "MinimumBaseAmount must be positive for %s %s", a, p)
+		assert.Positivef(t, limits.MaximumBaseAmount, "MaximumBaseAmount must be positive for %s %s", a, p)
+		assert.Positivef(t, limits.AmountStepIncrementSize, "AmountStepIncrementSize must be positive for %s %s", a, p)
```

Comment:
Can test all pairs:
```suggestion
		for _, p := range pairs {
			limits, err := b.GetOrderExecutionLimits(a, p)
			require.NoErrorf(t, err, "GetOrderExecutionLimits must not error for %s %s", a, p)
			assert.Positivef(t, limits.MinimumBaseAmount, "MinimumBaseAmount must be positive for %s %s", a, p)
			assert.Positivef(t, limits.MaximumBaseAmount, "MaximumBaseAmount must be positive for %s %s", a, p)
			assert.Positivef(t, limits.AmountStepIncrementSize, "AmountStepIncrementSize must be positive for %s %s", a, p)
		}
```

Reply:
Guess can do.
I was in the mood at this point where I want to minimise the pairs we want to trust from them, but I think that's a separate problem and I shouldn't project yesterday's frustrations onto today's PR.

Fixed 8f22ea3